### PR TITLE
Client Encryption : Add nuget info to Encryption csproj and compile against preview SDK

### PR DESCRIPTION
**Pull request by abhijitpai**
## Description
Make Encryption project depend on preview SDK.
Add nuget related attributes to Encryption.csproj

## Type of change
- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj [hunk 1]
```diff
@@ -4,15 +4,29 @@
     <AssemblyName>Microsoft.Azure.Cosmos.Encryption</AssemblyName>
     <RootNamespace>Microsoft.Azure.Cosmos.Encryption</RootNamespace>
     <LangVersion>latest</LangVersion>
+    <IsPreview>true</IsPreview>
+	
+    <CurrentDate>$([System.DateTime]::Now.ToString(yyyyMMdd))</CurrentDate>
+    <ClientVersion Condition=" '$(IsPreview)' != 'true' ">$(ClientOfficialVersion)</ClientVersion>
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:10]: Do the new package version align with the client version or its own?

Comment (abhijitpai) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:10]: Should it be 1.0-preview now?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:10]: Independent versioning makes sense.

**Comment by kirankumarkolli** [comment]
@j82w can you please guide package generation integration in official build. 

ref: https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/azure-pipelines-official.yml

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj [hunk 2]
```diff
@@ -4,15 +4,29 @@
...
+    <PackageLicenseUrl>https://aka.ms/netcoregaeula</PackageLicenseUrl>
+    <PackageProjectUrl>https://github.com/Azure/azure-cosmos-dotnet-v3</PackageProjectUrl>
+    <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288890</PackageIconUrl>
+    <PackageTags>microsoft;azure;cosmos;cosmosdb;documentdb;docdb;nosql;azureofficial;dotnetcore;netcore;netstandard;client;encryption;byok</PackageTags>
+
   </PropertyGroup>
+
   <ItemGroup>
-    <ProjectReference Include="..\..\Microsoft.Azure.Cosmos\src\Microsoft.Azure.Cosmos.csproj" />
+     <PackageReference Include="Microsoft.Azure.Cosmos" Version="$(Version)" />
```
Comment (j82w) [inline Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:27]: Any reason to not always target the latest? Locking it to one version has caused issues in the past where the SDK was updated with a fix but the version was locked so user's couldn't use it.

Comment (abhijitpai) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:27]: I haven't gone through nuget versioning in detail but I don't think our versioning currently aligns with what they expect which seems to be 3.8 < 3.9-preview < 3.9 whereas we have only 3.8 and 3.8-preview as of now which means nuget will end up picking 3.8 unless I force it to pick only previews, which is okay for now but still has the locking.

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj (hunk 1 above)
Comment (abhijitpai) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj]: Done.

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj (hunk 2 above)
Comment (abhijitpai) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj]: Made it *-preview for now.

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj [hunk 3]
```diff
@@ -4,15 +4,26 @@
...
+    <PackageTags>microsoft;azure;cosmos;cosmosdb;documentdb;docdb;nosql;azureofficial;dotnetcore;netcore;netstandard;client;encryption;byok</PackageTags>
   </PropertyGroup>
+
   <ItemGroup>
-    <ProjectReference Include="..\..\Microsoft.Azure.Cosmos\src\Microsoft.Azure.Cosmos.csproj" />
+     <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.8.0-preview" />  
   </ItemGroup>
 
-  <PropertyGroup>
-    <DefineConstants Condition=" '$(IsNightly)' == 'true' or '$(IsPreview)' == 'true' ">$(DefineConstants);PREVIEW</DefineConstants>
```
[the remark this answers is not in the document]
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:13]: Its preview only for now, but in futue when both SxS are needed, then above is needed.

[the remark this answers is not in the document]
Comment (abhijitpai) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj:13]: Added it back.

File: Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj (hunk 2 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Encryption/src/Microsoft.Azure.Cosmos.Encryption.csproj]: Deferring to GA timeline.

**Pull request merged**